From cb5f0329dd730dd6d86165ba9d2fd3e75c06c945 Mon Sep 17 00:00:00 2001
From: Vincent ABRIOU <vincent.abriou@st.com>
Date: Tue, 11 Feb 2020 10:06:13 +0100
Subject: [PATCH 4/6] TFLite: tools: make: add fftsg2d.c file in the build
 resources for libtensorflow-lite.a

lite/kernels/rfft2d.cc has reference to rdft2d.
As a consequence, libtensorflow-lite.a need to include fft2d/fftsg2d.c source
in its build.

If fftsg2d.c is not part of libtensorflow-lite.a, a C/C++ application
that use the libtensorflow-lite.a static library is not able to link
with the following error:
rfft2d.cc:(.text+0x594): undefined reference to `rdft2d'

Signed-off-by: Vincent ABRIOU <vincent.abriou@st.com>
---
 tensorflow/lite/tools/make/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tensorflow/lite/tools/make/Makefile b/tensorflow/lite/tools/make/Makefile
index bf973ea17a..c1affd75e1 100644
--- a/tensorflow/lite/tools/make/Makefile
+++ b/tensorflow/lite/tools/make/Makefile
@@ -112,6 +112,7 @@ $(wildcard tensorflow/lite/kernels/internal/reference/*.cc) \
 $(PROFILER_SRCS) \
 tensorflow/lite/tools/make/downloads/farmhash/src/farmhash.cc \
 tensorflow/lite/tools/make/downloads/fft2d/fftsg.c \
+tensorflow/lite/tools/make/downloads/fft2d/fftsg2d.c \
 tensorflow/lite/tools/make/downloads/flatbuffers/src/util.cpp
 CORE_CC_ALL_SRCS += \
 	$(shell find tensorflow/lite/tools/make/downloads/absl/absl/ \
-- 
2.17.1

